tests/components/lcn/test_device_trigger.py

```diff
@@ -0,0 +1,409 @@
+            for trigger in triggers:
+                assert trigger[CONF_TYPE] not in not_included_types
+
+
+@patch("pypck.connection.PchkConnectionManager", MockPchkConnectionManager)
+async def test_if_fires_on_transponder_event(hass, calls, entry):
+    """Test for transponder event triggers firing."""
+    await init_integration(hass, entry)
+    address = (0, 7, False)
+    device = get_device(hass, entry, address)
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        CONF_PLATFORM: "device",
+                        CONF_DOMAIN: DOMAIN,
+                        CONF_DEVICE_ID: device.id,
+                        CONF_TYPE: "transponder",
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {
+                            "test": "test_trigger_transponder",
+                            "code": "{{ trigger.event.data.code }}",
+                        },
+                    },
+                },
+            ]
+        },
+    )
+
+    inp = ModStatusAccessControl(
+        LcnAddr(*address),
+        periphery=AccessControlPeriphery.TRANSPONDER,
+        code="aabbcc",
+    )
+
+    lcn_connection = hass.data[DOMAIN][entry.entry_id][CONNECTION]
+    await lcn_connection.async_process_input(inp)
+    await hass.async_block_till_done()
+
+    assert len(calls) == 1
+    assert calls[0].data == {
+        "test": "test_trigger_transponder",
+        "code": "aabbcc",
+    }
+
+
+@patch("pypck.connection.PchkConnectionManager", MockPchkConnectionManager)
+async def test_if_fires_on_fingerprint_event(hass, calls, entry):
+    """Test for fingerprint event triggers firing."""
+    await init_integration(hass, entry)
+    address = (0, 7, False)
+    device = get_device(hass, entry, address)
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        CONF_PLATFORM: "device",
+                        CONF_DOMAIN: DOMAIN,
+                        CONF_DEVICE_ID: device.id,
+                        CONF_TYPE: "fingerprint",
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {
+                            "test": "test_trigger_fingerprint",
+                            "code": "{{ trigger.event.data.code }}",
+                        },
+                    },
+                },
+            ]
+        },
+    )
+
+    inp = ModStatusAccessControl(
+        LcnAddr(*address),
+        periphery=AccessControlPeriphery.FINGERPRINT,
+        code="aabbcc",
+    )
+
+    lcn_connection = hass.data[DOMAIN][entry.entry_id][CONNECTION]
```

Comment:
We don't want to access `hass.data` in the integration.  Instead access the mocked connection or patch the integration to get it.

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Reply:
You might save yourself some time by making `lcn_connection` a `pytest.fixture`

Reply:
I couldn't manage it this way. Instead I emulate the behavior of `unittest.Mock` and store the created instance in `return_value` of `MockPchkConnectionManager`.